From e0e624ba4c843a85ff0dfd7ea9692fc84bc2c597 Mon Sep 17 00:00:00 2001
From: Gowtham Anandha Babu <gowtham.anandha.babu@intel.com>
Date: Thu, 21 Dec 2023 16:43:02 +0530
Subject: [PATCH] Fix metadata issue when reconnect pixel

When reconnecting pixel some times metadata is not synced.

Issue: bt-stack issuing two AVRC browse channel, one as acceptor,
other as initator, which is causing collision in lcid and resulting
in l2cap disconnection of avrc browse channel, resulting in metadata
sync issue.

Solution: Don't initiate avrc browse channel as initiator, open only
as acceptor. Now browse channel is always success and metadata is
seen everytime.

Tracked-On: OAM-114183
Signed-off-by: Gowtham Anandha Babu <gowtham.anandha.babu@intel.com>
---
 bta/av/bta_av_act.cc | 9 +++++++--
 1 file changed, 7 insertions(+), 2 deletions(-)

diff --git a/bta/av/bta_av_act.cc b/bta/av/bta_av_act.cc
index 6f7a0e7c6..70e18aab7 100644
--- a/bta/av/bta_av_act.cc
+++ b/bta/av/bta_av_act.cc
@@ -581,8 +581,13 @@ void bta_av_rc_opened(tBTA_AV_CB* p_cb, tBTA_AV_DATA* p_data) {
   if ((p_cb->features & BTA_AV_FEAT_BROWSE) &&
       (rc_open.peer_features & BTA_AV_FEAT_BROWSE) &&
       ((p_cb->rcb[i].status & BTA_AV_RC_ROLE_MASK) == BTA_AV_RC_ROLE_INT)) {
-    APPL_TRACE_DEBUG("%s: opening AVRC Browse channel", __func__);
-    AVRC_OpenBrowse(p_data->rc_conn_chg.handle, AVCT_INT);
+    APPL_TRACE_DEBUG("%s: Not opening AVRC Browse channel as INT", __func__);
+    /* Commenting the below line as for IVI use case, for A2DP SNK, we dont
+     * support browsing channel. So no need to open browsing channel as
+     * INITIATOR. Revert this when have logic to resolve this conflict in
+     * stack/avct.
+     */
+    //AVRC_OpenBrowse(p_data->rc_conn_chg.handle, AVCT_INT);
   }
 }
 
-- 
2.17.1

